src/main/java/mods/eln/sixnode/AnalogChips.kt

```diff
@@ -234,33 +234,42 @@ class PIDRegulator : AnalogFunction() {
     internal var Ki = 0.0
         set(value) {
             field = value
-            errorIntegral = 0.0
+            integralTerm = 0.0
         }
     internal var Kd = 0.0
 
     private var lastError = 0.0
-    private var errorIntegral = 0.0
+    private var integralTerm  = 0.0
 
     override fun process(inputs: Array<Double?>, deltaTime: Double): Double {
         val error = (inputs[0] ?: 0.0) - (inputs[1] ?: 0.0)
-        errorIntegral += error * deltaTime
-        val result = Kp * error + Ki * errorIntegral + Kd * (error - lastError) / deltaTime
+
+        // proportional term calculation
+        val proportionalTerm = Kp * error
+
+        // integral term calculation
+        integralTerm += Ki * error * deltaTime
+        integralTerm = Utils.limit(integralTerm, -Eln.SVU, Eln.SVU) //windup protection
+
+        // derivative term calculation
+        val derivativeTerm = Kd * ((error - lastError) / deltaTime)
+
         lastError = error
-        return result
+        return Utils.limit(proportionalTerm + integralTerm + derivativeTerm, -Eln.SVU, Eln.SVU)
     }
 
     override fun readFromNBT(nbt: NBTTagCompound?, str: String?) {
         Kp = nbt?.getDouble("Kp") ?: 1.0
         Ki = nbt?.getDouble("Ki") ?: 0.0
         Kd = nbt?.getDouble("Kd") ?: 0.0
-        errorIntegral = nbt?.getDouble("errorIntegral") ?: 0.0
+        integralTerm = nbt?.getDouble("integralTerm") ?: 0.0
     }
 
     override fun writeToNBT(nbt: NBTTagCompound?, str: String?) {
         nbt?.setDouble("Kp", Kp)
         nbt?.setDouble("Ki", Ki)
         nbt?.setDouble("Kd", Kd)
-        nbt?.setDouble("errorIntegral", errorIntegral)
+        nbt?.setDouble("integralTerm", integralTerm)
     }
```

Comment:
This breaks backwards compat with worlds. Please revert or check for both.

Reply:
But the thing is that the new value has nothing to do with the old one, is it not bad when they are using the same base?

Reply:
You should generally try to make it so that if someone loads an old world, it will either take in the old configuration and make it fit the new one, or just use the old one. I don't see that here.

Reply:
Ok than I will change it.

Reply:
So I changed it, is it now good so?